user-impl/src/main/scala/com/example/auction/user/impl/UserServiceImpl.scala

```diff
@@ -2,38 +2,44 @@ package com.example.auction.user.impl
 
 import java.util.UUID
 
+import akka.NotUsed
 import akka.actor.ActorSystem
 import akka.persistence.cassandra.query.scaladsl.CassandraReadJournal
 import akka.persistence.query.PersistenceQuery
 import akka.stream.Materializer
 import akka.stream.scaladsl.Sink
 import com.example.auction.user.api
-import com.example.auction.user.api.UserService
+import com.example.auction.user.api._
 import com.lightbend.lagom.scaladsl.api.ServiceCall
 import com.lightbend.lagom.scaladsl.api.transport.NotFound
 import com.lightbend.lagom.scaladsl.persistence.PersistentEntityRegistry
 
 import scala.concurrent.ExecutionContext
 
-class UserServiceImpl(registry: PersistentEntityRegistry, system: ActorSystem)(implicit ec: ExecutionContext, mat: Materializer) extends UserService {
+class UserServiceImpl(registry: PersistentEntityRegistry, system: ActorSystem)
+                     (implicit ec: ExecutionContext, mat: Materializer) extends UserService {
 
   private val currentIdsQuery = PersistenceQuery(system).readJournalFor[CassandraReadJournal](CassandraReadJournal.Identifier)
 
-  override def createUser = ServiceCall { createUser =>
-    val userId = UUID.randomUUID()
-    refFor(userId).ask(CreateUser(createUser.name)).map { _ =>
-      api.User(userId, createUser.name)
+  override def createUser: ServiceCall[UserCreationRequest, api.User] =
+    ServiceCall { createUser =>
+      val userId = UUID.randomUUID()
+      refFor(userId)
+        .ask(CreateUser(createUser.username, createUser.email))
+        .map(_ => api.User(userId, createUser.username, createUser.email))
```

Comment:
I just remembered this is a mistake. If the request to create `User("ignasi-username", "ignasi@mail.com")` arrives twice then there'll be two users with the same username and e-mail and different `UUID`. To detect this we must decide one field in `CreateUser` to be the entity key (will need normalization --> toLowerCase? other?).

We can solve this issue in a separate PR with some tests and other stuff...

Reply:
I think we can make it simple by using the email as identifier.